bin/templates/cordova/lib/prepare.js

```diff
@@ -26,26 +26,43 @@ var AndroidManifest = require('./AndroidManifest');
 var xmlHelpers = require('cordova-common').xmlHelpers;
 var CordovaError = require('cordova-common').CordovaError;
 var ConfigParser = require('cordova-common').ConfigParser;
+var FileUpdater = require('cordova-common').FileUpdater;
 
-module.exports.prepare = function (cordovaProject) {
+module.exports.prepare = function (cordovaProject, options) {
 
     var self = this;
+    var platformResourcesDir = path.relative(cordovaProject.root, path.join(this.locations.root, 'res'));
 
     this._config = updateConfigFilesFrom(cordovaProject.projectConfig,
         this._munger, this.locations);
 
     // Update own www dir with project's www assets and plugins' assets and js-files
-    return Q.when(updateWwwFrom(cordovaProject, this.locations))
+    return Q.when(updateWww.call(self, cordovaProject))
     .then(function () {
         // update project according to config.xml changes.
         return updateProjectAccordingTo(self._config, self.locations);
     })
     .then(function () {
-        handleIcons(cordovaProject.projectConfig, self.root);
-        handleSplashes(cordovaProject.projectConfig, self.root);
+        updateIcons.call(self, cordovaProject, platformResourcesDir);
+        updateSplashes.call(self, cordovaProject, platformResourcesDir);
     })
     .then(function () {
-        events.emit('verbose', 'updated project successfully');
+        events.emit('verbose', 'Prepared project successfully');
+    });
+};
+
+module.exports.clean = function (options) {
+    // Unfortunately the cordovaProject isn't passed into the clean() function,
+    // but the project root dir and config can be resolved here easily.
```

Comment:
I guess I don't understand the non-CLI workflow very well. Why would 'cordova clean' be called in a non-CLI workflow, rather than 'gradle clean' or equivalent? Doesn't "non-CLI workflow" mean the cordova CLI commands like 'cordova clean' aren't being used?

In a normal cordova CLI workflow, I think users would expect that the files copied by prepare would be cleaned up by 'cordova clean' without any options; I was surprised to find they were not already.